File: 2.py
def buy_egg(tanggal, uang):
  jumlah_telur = uang // 2500
  bonus = 0
  for i in range(2, tanggal):
    if (tanggal % i) == 0:
      if (tanggal % 2) != 0:
        bonus = (jumlah_telur//12) * 3
    else:
      bonus = (jumlah_telur//12) * 2
  if (tanggal % 5) == 0:
    if (bonus % 2) == 0:
      bonus *= 10
    else:
      bonus *= 5
  print(jumlah_telur + bonus)

File: 2.js
function buyEgg(tanggal, uang){
  const jumlahTelur = uang / 2500;
  let bonusTelur = 0;
  for (let i=2; i<=tanggal; i++){
    if ((tanggal % i) === 0){
      if ((tanggal % 2) === 0){
        bonusTelur = Math.floor(jumlahTelur/12) * 3;
      }
    } else {
      bonusTelur = Math.floor(jumlahTelur/12) * 2;
    }
  } 
  if ((tanggal % 5) === 0){
    if ((tanggal % 2) === 0){
      bonusTelur *= 10;
    } else {
      bonusTelur *= 5;
    }
  }
  console.log(bonusTelur+jumlahTelur);
}
